assets/js/base/components/payment-methods/saved-payment-method-options.js

```diff
@@ -92,63 +96,100 @@ const SavedPaymentMethodOptions = ( { onSelect } ) => {
 		setActivePaymentMethod,
 	} = usePaymentMethodDataContext();
 	const [ selectedToken, setSelectedToken ] = useState( '' );
+	const standardMethods = getPaymentMethods();
+	const { selectedRates, shippingAddress } = useShippingDataContext();
+	const selectedShippingMethods = useShallowEqual( selectedRates );
+	const { cartTotals, cartNeedsShipping } = useStoreCart();
+	const canPayArgument = useRef( {
+		cartTotals,
+		cartNeedsShipping,
+		shippingAddress,
+		selectedShippingMethods,
+	} );
 
 	/**
 	 * @type      {Object} Options
 	 * @property  {Array}  current  The current options on the type.
 	 */
 	const currentOptions = useRef( [] );
-	useEffect( () => {
-		let options = [];
-		const paymentMethodKeys = Object.keys( customerPaymentMethods );
-		if ( paymentMethodKeys.length > 0 ) {
-			paymentMethodKeys.forEach( ( type ) => {
-				const paymentMethods = customerPaymentMethods[ type ];
-				if ( paymentMethods.length > 0 ) {
-					options = options.concat(
-						paymentMethods.map( ( paymentMethod ) => {
-							const option =
-								type === 'cc' || type === 'echeck'
-									? getCcOrEcheckPaymentMethodOption(
-											paymentMethod,
-											setActivePaymentMethod,
-											setPaymentStatus
-									  )
-									: getDefaultPaymentMethodOptions(
-											paymentMethod,
-											setActivePaymentMethod,
-											setPaymentStatus
-									  );
-							if (
-								paymentMethod.is_default &&
-								selectedToken === ''
-							) {
-								setSelectedToken( paymentMethod.tokenId + '' );
-								option.onChange( paymentMethod.tokenId );
-							}
-							return option;
-						} )
+	const updateOptions = useCallback( async () => {
+		// Admin users have all payment methods enabled, so we need to filter
+		// the ones that do not accept payments.
+		const getFilteredPaymentMethods = async ( paymentMethods ) => {
```

Comment:
All good points, I agree calling `canMakePayment` in two different places was not ideal. In fact that was what was making me not to be really comfortable with that approach either.

I took another take at this issue. Instead of filtering saved payment methods with `canMakePayment`, now I filter them with the `supports.savePaymentInfo` property, which we now set to false if the payment failed to register.

I also considered adding another attribute like `error` or `hasError` to 'mark' the payment methods that failed to register but are still added because the user is an admin. But I feel setting `savePaymentInfo` to false is a more natural approach.

In addition to that, I needed to replace the `content` so the error is thrown in the component (and then caught by the error boundary). Let me know if this new approach looks good to you.